integrations/huggingface-transformers/README.md

```diff
@@ -62,129 +96,207 @@ output:
 --dataset_name or --task_name: The dataset or task to load for training.
 ```
 
-## Sparse Transfer Learning | Question Answering Example
+Currently supported tasks include: 
+- `masked_language_modeling`
+- `text_classification`
+- `token_classification`
+- `question_answering`
+
+### SparseML Python API
+
+For additional flexibility, SparseML also offers a `Trainer` class that inherits from the familiar [Transformers's Trainer](https://huggingface.co/docs/transformers/main_classes/trainer). 
+
+SparseML's `Trainer` inherits all of the functionality from the Transformers repository, but also accepts a `recipe`, which allows you to specify sparsity-related algorithms and hyperparameters. The `Trainer` parses the recipe and adjusts the training loop to apply sparsification algorithms or sparse transfer learning. As such, you can leverage Hugging Face's friendly utilities such as the Model Hub, `AutoTokenizers`, `AutoModels`, and `datasets` in concert with SparseML's sparsity-related algorithms!
+
+We create the `Trainer` and kick of a run like this:
+
+```python
+from sparseml.transformers.sparsification import Trainer, TrainingArguments
+
+def run_training(model, model_path, recipe_path, teacher, training_args, dataset, tokenizer, compute_metrics):
+    # setup training loop based on recipe passed
+    trainer = Trainer(
+        recipe=recipe_path,
+        model=model,
+        model_state_path=model_path,
+        distill_teacher=teacher,
+        metadata_args=["per_device_train_batch_size","per_device_eval_batch_size","fp16"],
+        args=training_args,
+        train_dataset=dataset["train"],
+        eval_dataset=dataset["validation"],
+        tokenizer=tokenizer,
+        compute_metrics=compute_metrics
+    )
+
+    # run training
+    trainer.train(resume_from_checkpoint=False)
+```
+
+Check out the tutorials for actual working examples using the `Trainer` class.
```

Comment:
Check out the tutorials for actual ~~working~~ examples using the `Trainer` class